lib/Parse/ParseDecl.cpp

```diff
@@ -499,16 +755,74 @@ Parser::parseImplementsAttribute(SourceLoc AtLoc, SourceLoc Loc) {
                            ProtocolType.get(), MemberName, MemberNameLoc));
 }
 
+void Parser::parseObjCSelector(SmallVector<Identifier, 4> &Names,
+                               SmallVector<SourceLoc, 4> &NameLocs,
+                               bool &IsNullarySelector) {
```

Comment:
Nitpick: can we return `IsNullarySelector` instead of setting it? that way we'll never forget to set.

Reply:
Wouldn't it be weird to return some values by reference and one using return? Especially since `IsNullarySelector` is not in any way more important than `Names` and `NameLocs`.

Reply:
ah, That's fair. We can leave it as it is.